File: adapters/database/src/main/java/net/onelitefeather/pandorascluster/database/models/position/HomePositionEntity.java
package net.onelitefeather.pandorascluster.database.models.position;

import jakarta.persistence.*;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.util.Objects;

@Entity
@Table(name = "land_homes")
public final class HomePositionEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "world", length = 64, nullable = false)
    private String world;

    @Column(name = "pos_x", nullable = false)
    private Double posX;

    @Column(name = "pos_y", nullable = false)
    private Double posY;

    @Column(name = "pos_z", nullable = false)
    private Double posZ;

    @Column(name = "yaw", nullable = false)
    private Float yaw;

    @Column(name = "pitch", nullable = false)
    private Float pitch;

    public HomePositionEntity() {
        // Empty constructor for Hibernate
    }

    public HomePositionEntity(Long id, String world, Double posX, Double posY, Double posZ, Float yaw, Float pitch) {
        this.id = id;
        this.world = world;
        this.posX = posX;
        this.posY = posY;
        this.posZ = posZ;
        this.yaw = yaw;
        this.pitch = pitch;
    }

    public @Nullable Long id() {
        return id;
    }

    public @NotNull String world() {
        return world;
    }

    public double posX() {
        return posX;
    }

    public double posY() {
        return posY;
    }

    public double posZ() {
        return posZ;
    }

    public float yaw() {
        return yaw;
    }

    public float pitch() {
        return pitch;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof HomePositionEntity that)) return false;

        return Objects.equals(id, that.id) &&
                Objects.equals(world, that.world) &&
                Objects.equals(posX, that.posX) &&
                Objects.equals(posY, that.posY) &&
                Objects.equals(posZ, that.posZ) &&
                Objects.equals(yaw, that.yaw) &&
                Objects.equals(pitch, that.pitch);
    }

    @Override
    public int hashCode() {
        int result = Objects.hashCode(id);
        result = 31 * result + Objects.hashCode(world);
        result = 31 * result + Objects.hashCode(posX);
        result = 31 * result + Objects.hashCode(posY);
        result = 31 * result + Objects.hashCode(posZ);
        result = 31 * result + Objects.hashCode(yaw);
        result = 31 * result + Objects.hashCode(pitch);
        return result;
    }
}

File: adapters/database/src/main/kotlin/net/onelitefeather/pandorascluster/database/models/position/HomePositionEntity.kt
package net.onelitefeather.pandorascluster.database.models.position

import jakarta.persistence.*
import net.onelitefeather.pandorascluster.dbo.position.HomePositionDBO

@Entity
@Table(name = "land_homes")
data class HomePositionEntity(

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null,

    @Column
    val posX: Double = 1.0,

    @Column
    val posY: Double = 1.0,

    @Column
    val posZ: Double = 1.0,

    @Column
    val yaw: Float = 1.0F,

    @Column
    val pitch: Float = 1.0F
): HomePositionDBO {

    override fun hashCode(): Int = javaClass.hashCode()

    @Override
    override fun toString(): String =
        this::class.simpleName +
                "(id = $id , posX = $posX , posY = $posY , posZ = $posZ , " +
                "yaw = $yaw , pitch = $pitch)"

    override fun id(): Long? = id

    override fun posX(): Double = posX

    override fun posY(): Double = posY

    override fun posZ(): Double = posZ

    override fun yaw(): Float = yaw

    override fun pitch(): Float = pitch

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is HomePositionEntity) return false

        if (id != other.id) return false
        if (posX != other.posX) return false
        if (posY != other.posY) return false
        if (posZ != other.posZ) return false
        if (yaw != other.yaw) return false
        if (pitch != other.pitch) return false

        return true
    }
}